### DISCUSSION TITLE
Custom distance metric

### DISCUSSION THREAD

**Discussion by mattangus** [milvus-io/milvus#23045] [Q&A and General discussion]
I have a set of vectors and some learned shifts/scales for the cosine similarity query. In the below equation the `shift` and `scale` are uneque to that vector stored in the database.

```
sim[i] = (query.dot(vector[i]) + shift[i]) * scale[i]
```

Where `query` is the normalized query vector, `vector[i]` is the `i`th vector stored in the database. Is that possible with milvus?

**Comment by yhmo** [comment]
Nope, not support. Milvus calculates the cosine value or euclidean distance between vectors and returns the topk similar items. To support your requirement, we need some kind of user-define-function machinery to define an equation for similarity calculation.

**Comment by xiaofan-luan** [comment]
We should think of this seriously

**Comment by xiaofan-luan** [comment]
cc @liliu-z

**Comment by liliu-z** [comment]
Hi, that is definitely in our roadmap. I just opened a issue to track it https://github.com/milvus-io/milvus/issues/23112
